src/useConstant.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+import React from 'react'
+
+// This is dumb, but it's getting around a flow error, so...
+type Maybe<T> = T | void
+
+/**
+ * A simple hook to create a constant value that lives for
+ * the lifetime of the component
+ *
+ * Plagiarized from https://github.com/Andarist/use-constant
+ *
+ * @param {Function} init - A function to generate the value
+ */
+export default function useConstant<T>(init: () => T): T {
+  const ref = React.useRef<Maybe<T>>(undefined)
+  if (!ref.current) {
+    ref.current = init()
```

Comment:
just a note - I on purpose wrap this in an object and return its property so you technically can return falsy value from init and things still work as expected

Reply:
Yes. Yours is very general purpose in a way that I don't need mine to be. I'll add a comment.